iac-common/src/main/java/org/sonar/iac/common/api/tree/impl/TextRanges.java

```diff
@@ -51,6 +53,10 @@ public static TextRange merge(List<TextRange> ranges) {
     );
   }
 
+  public static TextRange mergeRanges(List<? extends HasTextRange> hasRanges) {
+    return hasRanges.isEmpty() ? null : merge(hasRanges.stream().map(HasTextRange::textRange).collect(Collectors.toList()));
+  }
```

Comment:
I think this will duplicate the methods Jonas has already implemented. When returning null you should annotate this utils method.